# Trabalho de ICD: Atividades Extraterrestres
Limpando o dataframe:

In [2]:
import pandas as pd
import numpy as np

In [4]:
import seaborn as sns
import matplotlib.pyplot as plt

In [24]:
df= pd.read_csv("./consolidated_weather_V03.csv")
df

state             city  colour      shape  mday  month  year  hour  \
0        WA          Olympia     NaN  Formation    14     11  1997    21   
1        WA          Olympia     NaN   Fireball    14     11  1997    21   
2        WA          Olympia     NaN  Formation    14     11  1997    21   
3        MD        Salisbury     NaN       Oval    24     11  1997     2   
4        MD        Salisbury  YELLOW       Oval    24     11  1997     2   
...     ...              ...     ...        ...   ...    ...   ...   ...   
51865    AZ  Prescott Valley     NaN     Circle    28      1  2000    20   
51866    CA           Covina     NaN      Light    27      1  2000    18   
51867    WA          Seattle     NaN   Fireball    26      1  2000    19   
51868    TX         Edinburg     NaN      Light    25      1  2000    19   
51869    NM       Rio Rancho     NaN     Sphere    23      1  2000    15   

            id  pressure  ...     dewpt  thunder  fog    precip      wspd  \
0          NaN -0.163149  ... -0.867472        0    0       NaN -0.357641   
1          NaN -0.163149  ... -0.867472        0    0       NaN -0.357641   
2          NaN -0.163149  ... -0.867472        0    0       NaN -0.357641   
3          NaN -0.189379  ... -0.464073        0    0       NaN -0.911239   
4          NaN -0.189379  ... -0.464073        0    0       NaN -0.911239   
...        ...       ...  ...       ...      ...  ...       ...       ...   
51865  57531.0  0.610616  ... -1.835629        0    0       NaN -0.357641   
51866  57536.0  0.413896  ...       NaN        0    0       NaN -0.911239   
51867  57543.0  1.751592  ... -0.595177        0    0       NaN -0.911239   
51868  57544.0  0.230291  ...  0.635190        0    0       NaN  0.373899   
51869  57549.0  0.545043  ... -2.481068        0    0 -0.092084  0.917612   

       tornado       hum  snow     wgust          conds  
0            0  1.598872     0       NaN          Clear  
1            0  1.598872     0       NaN          Clear  
2            0  1.598872     0       NaN          Clear  
3            0  0.980525     0       NaN  Mostly Cloudy  
4            0  0.980525     0       NaN  Mostly Cloudy  
...        ...       ...   ...       ...            ...  
51865        0 -0.731823     0       NaN          Clear  
51866        0       NaN     0       NaN          Clear  
51867        0  0.790264     0       NaN  Mostly Cloudy  
51868        0  0.171916     0       NaN       Overcast  
51869        0 -2.539302     0 -0.640525          Clear  

[51870 rows x 27 columns]

O comando .info() nos trás uma descrição de cada coluna e assim podemos decidir com quias vamos trabalhar e se é possivel colunas e deixar o dataframe menor. Vamos analisar as colunas e depois limpar as linhas. 

In [25]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51870 entries, 0 to 51869
Data columns (total 27 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   state      51870 non-null  object 
 1   city       51870 non-null  object 
 2   colour     80 non-null     object 
 3   shape      51868 non-null  object 
 4   mday       51870 non-null  int64  
 5   month      51870 non-null  int64  
 6   year       51870 non-null  int64  
 7   hour       51870 non-null  int64  
 8   id         51631 non-null  float64
 9   pressure   51659 non-null  float64
 10  temp       51544 non-null  float64
 11  hail       51870 non-null  int64  
 12  heatindex  3589 non-null   float64
 13  windchill  8718 non-null   float64
 14  rain       51870 non-null  int64  
 15  vis        51170 non-null  float64
 16  wdire      51863 non-null  object 
 17  dewpt      51393 non-null  float64
 18  thunder    51870 non-null  int64  
 19  fog        51870 non-null  int64  
 20  precip

In [26]:
df[['rain']].describe()

rain
count  51870.000000
mean       0.021477
std        0.144969
min        0.000000
25%        0.000000
50%        0.000000
75%        0.000000
max        1.000000

In [27]:
df["hail"].value_counts()

hail
0    51870
Name: count, dtype: int64

A parti de dois comandos .info() e .volue_counts() é possivel visualizar que a coluna não tem dado. Sendo assim podemos removelo do nosso dataframe. 

In [28]:
df = df.drop('hail', axis = 1)

In [29]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51870 entries, 0 to 51869
Data columns (total 26 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   state      51870 non-null  object 
 1   city       51870 non-null  object 
 2   colour     80 non-null     object 
 3   shape      51868 non-null  object 
 4   mday       51870 non-null  int64  
 5   month      51870 non-null  int64  
 6   year       51870 non-null  int64  
 7   hour       51870 non-null  int64  
 8   id         51631 non-null  float64
 9   pressure   51659 non-null  float64
 10  temp       51544 non-null  float64
 11  heatindex  3589 non-null   float64
 12  windchill  8718 non-null   float64
 13  rain       51870 non-null  int64  
 14  vis        51170 non-null  float64
 15  wdire      51863 non-null  object 
 16  dewpt      51393 non-null  float64
 17  thunder    51870 non-null  int64  
 18  fog        51870 non-null  int64  
 19  precip     2011 non-null   float64
 20  wspd  

Agora vamos remover as linhas duplicadas do dataframe, usando o comando .drop_duplicates())

In [33]:
df = df.drop_duplicates()

In [34]:
df

state             city  colour      shape  mday  month  year  hour  \
0        WA          Olympia     NaN  Formation    14     11  1997    21   
1        WA          Olympia     NaN   Fireball    14     11  1997    21   
3        MD        Salisbury     NaN       Oval    24     11  1997     2   
4        MD        Salisbury  YELLOW       Oval    24     11  1997     2   
5        MS         Gulfport     NaN   Triangle    31     12  1997    17   
...     ...              ...     ...        ...   ...    ...   ...   ...   
51865    AZ  Prescott Valley     NaN     Circle    28      1  2000    20   
51866    CA           Covina     NaN      Light    27      1  2000    18   
51867    WA          Seattle     NaN   Fireball    26      1  2000    19   
51868    TX         Edinburg     NaN      Light    25      1  2000    19   
51869    NM       Rio Rancho     NaN     Sphere    23      1  2000    15   

            id  pressure  ...     dewpt  thunder  fog    precip      wspd  \
0          NaN -0.163149  ... -0.867472        0    0       NaN -0.357641   
1          NaN -0.163149  ... -0.867472        0    0       NaN -0.357641   
3          NaN -0.189379  ... -0.464073        0    0       NaN -0.911239   
4          NaN -0.189379  ... -0.464073        0    0       NaN -0.911239   
5          NaN  2.105688  ... -0.968322        0    0       NaN  0.373899   
...        ...       ...  ...       ...      ...  ...       ...       ...   
51865  57531.0  0.610616  ... -1.835629        0    0       NaN -0.357641   
51866  57536.0  0.413896  ...       NaN        0    0       NaN -0.911239   
51867  57543.0  1.751592  ... -0.595177        0    0       NaN -0.911239   
51868  57544.0  0.230291  ...  0.635190        0    0       NaN  0.373899   
51869  57549.0  0.545043  ... -2.481068        0    0 -0.092084  0.917612   

      tornado       hum  snow     wgust          conds  
0           0  1.598872     0       NaN          Clear  
1           0  1.598872     0       NaN          Clear  
3           0  0.980525     0       NaN  Mostly Cloudy  
4           0  0.980525     0       NaN  Mostly Cloudy  
5           0 -1.207476     0       NaN  Partly Cloudy  
...       ...       ...   ...       ...            ...  
51865       0 -0.731823     0       NaN          Clear  
51866       0       NaN     0       NaN          Clear  
51867       0  0.790264     0       NaN  Mostly Cloudy  
51868       0  0.171916     0       NaN       Overcast  
51869       0 -2.539302     0 -0.640525          Clear  

[51830 rows x 26 columns]

Nosso dataframe tinha 51870 linhas e agora temos 51830 linhas, ou aparições.

In [38]:
df.to_csv('novodf.csv')